### ISSUE TITLE
Make diff for failed assertions in IDEA IDE

### ISSUE THREAD

**Issue by kkocel** [kotest/kotest#4435]
I would like to have similar functionality as Junit 5 has in IDEA.

When assertion fails, there is an option to click a button to see how objects asserted are different:

<img width="285" alt="image" src="https://github.com/user-attachments/assets/8c4fb706-b2c8-4f7b-8f4b-304c9ccf468a">
 
In kotest I get detailed information about objects but no easy option to see the diff.

```
java.lang.AssertionError: Collection should have size 2 but has size 1. Values: [...]
	at com.example.SomeTest.should do sth
```

**Comment by AlexCue987** [comment]
do you have kotest plugin installed ?

**Comment by kkocel** [comment]
Yes

**Comment by Nikita-Guchakov** [comment]
The problem does not related to the plugin, it's about throwing `io.kotest.assertions.AssertionFailedError` instead of `java.lang.AssertionError` for assertions. IDE supports [opentest4j](https://github.com/ota4j-team/opentest4j) initiative, and `AssertionFailedError` extends from `org.opentest4j.AssertionFailedError`. So fixing assertions to throw proper exception will be enough and will work without plugin either.

Same issue as in the [comment](https://github.com/kotest/kotest/issues/4500#issuecomment-2558357818)

**Comment by kkocel** [comment]
@Nikita-Guchakov I've seen that kotest is using those exceptions [already](https://github.com/kotest/kotest/pull/1339/files), maybe it's a bug that it is not used here...

**Comment by kkocel** [comment]
@LeoColman so it's not intellij-plugin issue, it's opentest4j regression

**Comment by Kantis** [comment]
What actual diff do you wish to see here?

assuming `listOf("foo") shouldHaveSize 2` fails, what is a meaningful _diff_?

**Comment by kkocel** [comment]
I would rather be able to leverage [opentest4j](https://github.com/ota4j-team/opentest4j) support in kotest, the assertion I put in the description is a sample.

**Comment by sksamuel** [comment]
I don't think this is an opentest4j thing - the diff is generated by formatting the message in a certain way in the output. We must be doing that wrong somewhere.

**Comment by Nikita-Guchakov** [comment]
@sksamuel opentest4j format is supported by IDEs, it works properly everywhere kotest uses it. Just switching to opentest4j there as well should work for this case too

**Comment by sksamuel** [comment]
It's because some of our assertions don't include the actual/expected values. All cases of `x shouldBe y` do, but lots of the custom ones don't (yet). I've fixed all size comparisons in this PR so this will be fixed for 6.0
